simulation/simple_sensor_simulator/src/vehicle_simulation/ego_entity_simulation.cpp

```diff
@@ -204,10 +204,21 @@ void EgoEntitySimulation::update(
 {
   autoware->rethrow();
 
+  auto getTargetSpeed = [&]() -> double {
+    if (target_speed_.has_value()) {
+      return target_speed_.value();
+    } else if (status_.lanelet_pose_valid) {
+      return std::min(
+        hdmap_utils_ptr_->getSpeedLimit(
+          hdmap_utils_ptr_->getFollowingLanelets(status_.lanelet_pose.lanelet_id)),
+        behavior_parameter_.dynamic_constraints.max_speed);
+    } else {
+      return behavior_parameter_.dynamic_constraints.max_speed;
+    }
+  };
```

Comment:
It is not a good idea to use the upper speed limit embedded in the lanelet as the target speed for FollowTrajectoryAction, which has no speed specified by SpeedAction and no arrival time to a waypoint. This is because FollowTrajectoryAction can also travel outside of the lane, and this idea is not an answer for such cases. There is also the question of whether the speed limit of the lane can be used as the target speed when moving across the lane.

I considered this for a while, but could not come up with a speed that could be adopted as a reliable target speed in such a case. Therefore, in such cases, the speed at the time the FollowTrajectoryAction starts should be used as the target speed until the FollowTrajectoryAction ends (when there is no target speed by SpeedAction or arrival time specification). If doing so eliminates the need to send and receive new message types via ZeroMQ (as this pull request does), then that is best.

Reply:
> Therefore, in such cases, **the speed at the time** the FollowTrajectoryAction **starts should be used as the target speed** until the FollowTrajectoryAction ends (when there is no target speed by SpeedAction or arrival time specification). If doing so eliminates the need to send and receive new message types via ZeroMQ (as this pull request does), then that is best.

You mean that in such a case (calling `requestFollowTrajectory` for Ego with Autoware) should the current linear speed be read and set as `target_speed` in `FollowTrajectoryAction`? 
What if `target_speed` has been previously set by `SpeedActionTarget` - then I should leave the current value  (from `SpeedActionTarget`) unchanged?

Reply:
_NOTE: See https://github.com/tier4/scenario_simulator_v2/pull/1163#discussion_r1452927183._